rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -2233,6 +2292,247 @@ public boolean hasBtcBlockCoinbaseTransactionInformation(Sha256Hash blockHash) {
+        }
+
+        Context.propagate(btcContext);
+        Sha256Hash btcTxHash = BtcTransactionFormatUtils.calculateBtcTxHash(btcTxSerialized);
+
+        Keccak256 fastBridgeDerivationHash = getFastBridgeDerivationHash(
+                derivationArgumentsHash,
+                userRefundAddress,
+                lpBtcAddress,
+                lbcAddress
+        );
+
+        if (provider.isFastBridgeFederationDerivationHashUsed(btcTxHash, fastBridgeDerivationHash)) {
+            logger.debug("[registerFastBridgeBtcTransaction] Transaction and derivation hash already used");
+            return FAST_BRIDGE_UNPROCESSABLE_TX_ALREADY_PROCESSED_ERROR_CODE;
+        }
+
+        if (!validationsForRegisterBtcTransaction(btcTxHash, height, pmtSerialized, btcTxSerialized)) {
+            logger.debug(
+                "[registerFastBridgeBtcTransaction] (btcTx:{}) error during validationsForRegisterBtcTransaction",
+                btcTxHash
+            );
+            return FAST_BRIDGE_UNPROCESSABLE_TX_VALIDATIONS_ERROR;
+        }
+
+        BtcTransaction btcTx = new BtcTransaction(bridgeConstants.getBtcParams(), btcTxSerialized);
+        btcTx.verify();
+
+        Sha256Hash btcTxHashWithoutWitness = btcTx.getHash(false);
+        if (!btcTxHashWithoutWitness.equals(btcTxHash) &&
+            (provider.isFastBridgeFederationDerivationHashUsed(btcTxHashWithoutWitness, derivationArgumentsHash))) {
+            logger.debug("[registerFastBridgeBtcTransaction] Transaction and derivation hash already used");
+            return FAST_BRIDGE_UNPROCESSABLE_TX_ALREADY_PROCESSED_ERROR_CODE;
+        }
+
+        FastBridgeFederationInformation fastBridgeFederationInformation =
+            createFastBridgeFederationInformation(fastBridgeDerivationHash);
+
+        Address fastBridgeFedAddress =
+            fastBridgeFederationInformation.getFastBridgeFederationAddress(bridgeConstants.getBtcParams());
+
+        Coin totalAmount = getAmountSentToAddress(btcTx, fastBridgeFedAddress);
+
+        if (totalAmount == Coin.ZERO) {
+            logger.debug("[registerFastBridgeBtcTransaction] Amount sent can't be 0");
+            return FAST_BRIDGE_UNPROCESSABLE_TX_VALUE_ZERO_ERROR;
+        }
+
+        if (!verifyLockDoesNotSurpassLockingCap(btcTx, totalAmount)) {
+            InternalTransaction internalTx = (InternalTransaction)rskTx;
+            logger.info("[registerFastBridgeBtcTransaction] Locking cap surpassed, going to return funds!");
+            WalletProvider walletProvider = createFastBridgeWalletProvider(fastBridgeFederationInformation);
+            if (shouldTransferToContract) {
+                logger.debug("[registerFastBridgeBtcTransaction] Returning to liquidity provider");
+                generateRejectionRelease(btcTx, lpBtcAddress, fastBridgeFedAddress, new Keccak256(internalTx.getOriginHash()), totalAmount, walletProvider);
+                return FAST_BRIDGE_REFUNDED_LP_ERROR_CODE;
```

Comment:
Seems that you are returning before saving the btc hash as processed. This could allow a user to 1. recover the funds by forcing this rejectiond and 2. later registering the btc tx regularly.

It'd be nice to have a test that first rejects a transaction because it surprassed the locking cap and later it rejects the transaction because it was already processed